pkg/wallet/wallet.go

```diff
@@ -123,8 +130,36 @@ func (w *BaseWallet) Close() error {
 }
 
 // CreateDID returns new DID Document
-func (w *BaseWallet) CreateDID() (*did.Doc, error) {
-	return nil, fmt.Errorf("To be implemented")
+func (w *BaseWallet) CreateDID(method string) (*did.Doc, error) {
+	// Generate key pair
+	pub, err := w.CreateKey()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create DID: %w", err)
+	}
+
+	// DID identifier
+	id := fmt.Sprintf(didFormat, method, pub[:16])
+
+	// Supporting only one public key now
+	pubKey := did.PublicKey{
+		ID: fmt.Sprintf(didPKID, id, 1),
+		//TODO hardcoding public key type for now
+		// Should be dynamic for multi-key support
+		Type:       "Ed25519VerificationKey2018",
+		Controller: id,
+		Value:      []byte(pub),
+	}
+
+	// Created time
+	createdTime := time.Now()
+
+	return &did.Doc{
```

Comment:
You are missing service endpoint

Reply:
done

Reply:
done

Reply:
Also the DID should be saved to the store.

Reply:
We decided not to persist DID for now when we discussed last time, and once need comes we can add function to retrieve DID as well from store.